### PR TITLE
[processor/awsapplicationsignalsprocessor] Create No-Op processor skeleton for AWS Application Signals Processor

### PR DESCRIPTION
**Pull request by jj22ee**
**Description:** <Describe what has changed.>

The PR adds an initial skeleton implementation of the `awsapplicationsignalsprocessor`, which is currently a No-Op implementation for the processor for traces+metrics.
- The intention of the AWS Application Signals processor is used to reduce the cardinality of telemetry metrics and traces before exporting them to CloudWatch Logs via [EMF](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsemfexporter) and [X-Ray](github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter) respectively.

A series of PRs will follow for the full implementation of this new processor.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32808

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/awsapplicationsignalsprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# AWS Application Signals Processor
+
+The AWS Application Signals processor is used to reduce the cardinality of telemetry metrics and traces before exporting them to CloudWatch Logs via [EMF](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsemfexporter) and [X-Ray](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsxrayexporter) respectively.
+It reduces the cardinality of metrics/traces via 3 types of actions, `keep`, `drop` and `replace`, which are configured by users. Users can configure these rules through configurations.
```
Comment (tiffany76) [inline processor/awsapplicationsignalsprocessor/README.md:4]:
```suggestion
It reduces the cardinality of metrics and traces using three types of actions, `keep`, `drop` and `replace`, which are configured by users. Users can configure these rules through configurations.
```

File: processor/awsapplicationsignalsprocessor/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Note: Traces support only `replace` actions and are implicitly pulled from the logs section of the configuration
```
Comment (tiffany76) [inline processor/awsapplicationsignalsprocessor/README.md:6]:
```suggestion
Note: Traces support only `replace` actions and are implicitly pulled from the logs section of the configuration.
```

**Review by tiffany76 — COMMENTED** [review]
Docs reviewer here. Just a few small style suggestions. Thanks!

File: processor/awsapplicationsignalsprocessor/README.md (hunk 1 above)
Comment (jj22ee) [reply processor/awsapplicationsignalsprocessor/README.md]: Thanks, committed suggestion.

File: processor/awsapplicationsignalsprocessor/README.md (hunk 2 above)
Comment (jj22ee) [reply processor/awsapplicationsignalsprocessor/README.md]: Committed suggestion!

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -158,6 +158,7 @@ pkg/translator/zipkin/                                              @open-teleme
 pkg/winperfcounters/                                                @open-telemetry/collector-contrib-approvers @dashpole @Mrod1598 @BinaryFissionGames @alxbl
 
 processor/attributesprocessor/                                      @open-telemetry/collector-contrib-approvers @boostchicken
+processor/awsapplicationsignalsprocessor/                           @open-telemetry/collector-contrib-approvers @Aneurysm9
```
Comment (mxiamxia) [inline .github/CODEOWNERS:161]: could you add me in the code reviewer/approver list as well? I will also take the responsibilities to contribute and address the issues for this processor.

File: processor/awsapplicationsignalsprocessor/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# AWS Application Signals Processor
+
+The AWS Application Signals processor is used to reduce the cardinality of telemetry metrics and traces before exporting them to CloudWatch Logs via [EMF](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsemfexporter) and [X-Ray](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsxrayexporter) respectively.
+It reduces the cardinality of metrics and traces using three types of actions, `keep`, `drop` and `replace`, which are configured by users. Users can configure these rules through configurations.
+
+Note: Traces support only `replace` actions and are implicitly pulled from the logs section of the configuration.
```
Comment (mxiamxia) [inline processor/awsapplicationsignalsprocessor/README.md:6]: Could you please include descriptions of the four functioning components for the processor mentioned in this link: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32808?

File: .github/CODEOWNERS (hunk 3 above)
Comment (jj22ee) [reply .github/CODEOWNERS]: Added.

**Review by mxiamxia — APPROVED** [review]

**Comment by jj22ee** [comment]
Hi @MovieStoreGuy, can you help please approve/merge this PR?
We don't have a new sponsor as our previous sponsor is gone, but we have approvals from AWS ADOT team.

**Review by wangzlei — APPROVED** [review]

File: processor/awsapplicationsignalsprocessor/factory.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+const (
+	// The stability level of the processor.
+	stability = component.StabilityLevelBeta
```
Comment (MovieStoreGuy) [inline processor/awsapplicationsignalsprocessor/factory.go:17]: This should be `Development` while it is performing a NoOp

File: processor/awsapplicationsignalsprocessor/factory.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+var (
+	// The value of "type" key in configuration.
+	typeStr, _           = component.NewType("awsapplicationsignals")
```
Comment (MovieStoreGuy) [inline processor/awsapplicationsignalsprocessor/factory.go:22]: Please use `component.MustNewType` instead here to ensure that it matches the expected value.

File: processor/awsapplicationsignalsprocessor/processor.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type awsapplicationsignalsprocessor struct{}
+
+func (ap *awsapplicationsignalsprocessor) StartMetrics(_ context.Context, _ component.Host) error {
```
Comment (MovieStoreGuy) [inline processor/awsapplicationsignalsprocessor/processor.go:16]: Is there any particular reason why `StartMetrics` and `StartTraces` are not combined as `Start` ?

**Review by MovieStoreGuy — COMMENTED** [review]
Just some few minor nits as glance through, you will need to find a sponsor as part of the issue raised for this PR.

File: processor/awsapplicationsignalsprocessor/processor.go (hunk 7 above)
Comment (jj22ee) [reply processor/awsapplicationsignalsprocessor/processor.go:16]: When a new processor is created via a factory (via `NewFactory()`), providing processor logic for traces and metrics is done separately (via [processor.WithTraces and processor.WithMetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/870e99509a62f720601c93b6c4dcd5b4be29135b/processor/awsapplicationsignalsprocessor/factory.go#L31-L32))

So, `StartMetrics` and `StartTraces` are for the respective `processorhelper.NewMetricsProcessor` and `processorhelper.NewTracesProcessor`, for the respective metrics and traces processor.

File: processor/awsapplicationsignalsprocessor/factory.go (hunk 5 above)
Comment (jj22ee) [reply processor/awsapplicationsignalsprocessor/factory.go]: Changed to `StabilityLevelDevelopment`

File: processor/awsapplicationsignalsprocessor/factory.go (hunk 6 above)
Comment (jj22ee) [reply processor/awsapplicationsignalsprocessor/factory.go]: Changed to `component.MustNewType`.

**Pull request closed without merging**